src/test/ui/deriving/deriving-all-codegen.stdout

```diff
@@ -481,6 +486,99 @@ impl<T: ::core::cmp::Ord, U: ::core::cmp::Ord> ::core::cmp::Ord for
     }
 }
 
+// A packed, generic tuple struct. Because it is packed, a `T: Copy` bound is
+// added to all impls except for `Default`. This is because we must access
+// fields using copies (e.g. `&{self.0}`), instead of using direct references
+// (e.g. `&self.0`) which may be undefined in a packed struct.
+#[repr(packed)]
+struct PackedGeneric<T>(T, u32);
+#[automatically_derived]
+impl<T: ::core::clone::Clone + ::core::marker::Copy> ::core::clone::Clone for
+    PackedGeneric<T> {
+    #[inline]
+    fn clone(&self) -> PackedGeneric<T> {
+        PackedGeneric(::core::clone::Clone::clone(&{ self.0 }),
+            ::core::clone::Clone::clone(&{ self.1 }))
+    }
+}
+#[automatically_derived]
+impl<T: ::core::marker::Copy> ::core::marker::Copy for PackedGeneric<T> { }
+#[automatically_derived]
+impl<T: ::core::fmt::Debug + ::core::marker::Copy> ::core::fmt::Debug for
+    PackedGeneric<T> {
+    fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
+        ::core::fmt::Formatter::debug_tuple_field2_finish(f, "PackedGeneric",
+            &&{ self.0 }, &&{ self.1 })
```

Comment:
Strange to see `&&` here but it's the same without `packed` :shrug:

Reply:
I thought that was necessary because `debug_tuple_field2_finish` takes a `&dyn Debug`, but seemingly not. I'll address that in a different PR.

Reply:
Now I remember: it's for fields with unsized types. E.g. in `struct Unsized([u32])` you have to pass `&&self.0` to the function expecting `&dyn Debug`. And `&&` works for sized types as well, so we always do it even though it's not needed for sized types, because it's not worth the effort to avoid.